memq-client/src/main/java/com/pinterest/memq/client/commons2/network/NetworkClient.java

```diff
@@ -218,12 +228,39 @@ public boolean isClosed() {
 
   // blocking
   public void reset() throws IOException, InterruptedException {
-    if (connectFuture != null && connectFuture.channel() != null) {
-      connectFuture.channel().close().await(); // should reset the response map after channel is closed
-      connectFuture = null;
+    if (activeConnection != null && activeConnection.channel() != null) {
+      activeConnection.channel().close().await(); // should reset the response map after channel is closed
+      activeConnection = null;
     }
   }
 
+  public void drain() {
+    if (activeConnection != null && activeConnection.channel() != null) {
```

Comment:
Needs additional documentation and minor fix that drain doesn't create new connection.

What's the expected case when activeConnection is null and drain is invoked?